mlrun/db/base.py

```diff
@@ -142,6 +143,85 @@ def delete_function(self, name: str, project: str = ""):
     def list_functions(self, name=None, project="", tag="", labels=None):
         pass
 
+    @abstractmethod
+    def tag_objects(
+        self,
+        project: str,
+        tag_name: str,
+        tag_objects: schemas.TagObjects,
+        replace: bool = False,
+    ):
+        pass
+
+    def delete_objects_tag(
```

Comment:
You implemented the methods in all the inheriting classes, so you can (and should) set these as abstract methods
```suggestion
    @abstractmethod
    def delete_objects_tag(
```